shared/actions/search.js

```diff
@@ -223,3 +205,27 @@ export function reset (): Reset {
     payload: {},
   }
 }
+
+function * handleSearchRequest (action: Search): SagaGenerator<any, any> {
+  if (action.error) {
+    return
+  }
+
+  const {term, platform} = action.payload
+  try {
+    const results: APIRes = yield * cpsWithWaiting(waiting, _search, term, platform)
+    const json = JSON.parse(results.body)
+    const followingState = yield select((state: TypedState) => state.config.following) // eslint-disable-line arrow-parens
```

Comment:
Is using select here something that should be avoided? Trying to think about what the alternative would look like: we'd pass in the followingState and username as arguments to the saga? Is doing it that way cleaner (so that the the state doesn't get out of sync wrt the saga start)?